includes/admin/reporting/tabs/class.llms.admin.reporting.tab.students.php

```diff
@@ -94,13 +94,20 @@ public function breadcrumbs() {
 	 *
 	 * @return   void
 	 * @since    3.2.0
+	 * @since    [version] Added a check to see if the given user exists before outputting the report template.
 	 * @version  3.2.0
 	 */
 	public function output() {
 
 		// Single student.
 		if ( isset( $_GET['student_id'] ) ) {
 
+			llms_get_student( $_GET['student_id'] );
+			$student = llms_get_student( $_GET['student_id'] );
+			if ( ! $student ) {
+				wp_die( __( 'You do not have permission to access this user.', 'lifterlms' ) );
```

Comment:
I honestly don't think this error message fits.
I mean, this happens literally when the student doesn't exist. What do you think @pondermatic?

Reply:
I made the message similar to the one in the [courses report](https://github.com/gocodebox/lifterlms/blob/b5a476bf660363117cb8130caccc931f13e4b6e1/includes/admin/reporting/tabs/class.llms.admin.reporting.tab.courses.php#L99)

Reply:
Ah, I see, well in that case though there's a check on the current user's capabilities so it makes more sense...
Even if I agree that a non existent course id passed there will still produce the "insufficient permissions" error message.

You know what? we could do this instead:
```php
$student_id = llms_filter_input( INPUT_GET, 'student_id', FILTER_SANITIZE_NUMBER_INT );
if ( llms_current_user_can( 'view_lifterlms_reports', $student_id ) ) {
	wp_die( __( 'You do not have permission to access this user.', 'lifterlms' ) );
}
$student = llms_get_student( $student_id );
```
how does that look for you? Does it work fine?

Reply:
That doesn't work because `llms_current_user_can()` will grant access if the user has the `view_others_lifterlms_reports` capability.

Reply:
Nice, combination of them so, what do you think?
```php
$student_id = llms_filter_input( INPUT_GET, 'student_id', FILTER_SANITIZE_NUMBER_INT );
$student    = llms_get_student( $student_id );
if ( ! $student ) {
	wp_die( __( "This student doesn't exist.", 'lifterlms' ) );
}
if ( llms_current_user_can( 'view_lifterlms_reports', $student_id ) ) {
	wp_die( __( "You do not have permission to access this student's reports", 'lifterlms' ) );
}
```
or just IMHO, your initial version, but with a message that actually says what's happening: the student doesn't exist.
Do you agree?
Or am I missing something, like that you don't want to reveal this information for some reason?

Reply:
I usually don't reveal that something doesn't exist so that an attacker doesn't know why the request failed. Unfortunately, it might confuse honest users. So I could go either way on this message.

Reply:
I'd like it :)

Reply:
:+1: